### DISCUSSION TITLE
QtWebEngine buildtree

### DISCUSSION THREAD

**Discussion by Melio-1** [microsoft/vcpkg#40236] [Q&A] [answered]
Why when install QtWebEngine say's Buildtree path 'P:/dev/vcpkg/buildtrees/qtwebengine' is too long.

**Comment by autoantwort** [comment]
It is a msvc limitation: https://developercommunity.visualstudio.com/t/compiler-cant-find-source-file-in-path/10221576

Add `--x-buildtrees-root=P:/dev/b` to workaround this issue.

**Reply by Melio-1** [reply]
> It is a msvc limitation: https://developercommunity.visualstudio.com/t/compiler-cant-find-source-file-in-path/10221576
> 
> Add `--x-buildtrees-root=P:/dev/b` to workaround this issue.

i tried `--x-buildtrees-root=P:/dev/QT` and didn't work unfortunately

**Reply by autoantwort** [reply]
And `--x-buildtrees-root=P:/b`?

**Reply by Melio-1** [reply]
> And `--x-buildtrees-root=P:/b`?

i well try it after updating

**Reply by Melio-1** [reply]
> And `--x-buildtrees-root=P:/b`?
Same problem
![Screenshot 2024-08-09 230558](https://github.com/user-attachments/assets/b659f79e-302f-4bdb-ab26-e02d0335beba)

**Reply by Osyotr** [reply]
Do you really need all of qt? `qtbase` might be enough for you.

**Reply by Melio-1** [reply]
> Do you really need all of qt? `qtbase` might be enough for you.

No, qtbase is enough but i need to know why give me this bulid error

**Reply by Neumann-A** [reply]
What is the exact error ?

**Reply by Melio-1** [reply]
> What is the exact error ?

see issue #40269

**Reply by Neumann-A** [reply, marked as answer]
```
INTERNAL COMPILER ERROR in 'P:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.40.33807\bin\Hostx64\x64\cl.exe'
    Please choose the Technical Support command on the Visual C++
    Help menu, or open the Technical Support help file for more information
```
ICE compiler problem. Nothing you could do except using a different compiler version

**Reply by Melio-1** [reply]
> ```
> INTERNAL COMPILER ERROR in 'P:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.40.33807\bin\Hostx64\x64\cl.exe'
>     Please choose the Technical Support command on the Visual C++
>     Help menu, or open the Technical Support help file for more information
> ```
> 
> ICE compiler problem. Nothing you could do except using a different compiler version
So if i install different verdion of compiler will work? like 14.39?

**Reply by autoantwort** [reply]
You have to test it

**Reply by Melio-1** [reply]
> You have to test it

Ok, i will feedback if it's work or not

**Reply by Melio-1** [reply]
> You have to test it

i have a question i had install SDL2 successfully but when i tried to include in my project say's file not found, i'm using vs code and cmake

**Reply by autoantwort** [reply]
Without code + error log I can only guess

**Reply by Melio-1** [reply]
> Without code + error log I can only guess

I Get this message:


Building Custom Rule P:/Users/MEL/Desktop/Programming/CPP/c++11/CMakeLists.txt
[build]   main.cpp
[build] MSVCRTD.lib(exe_main.obj) : error LNK2019: unresolved external symbol main referenced in function "int __cdecl invoke_main(void)" (?invoke_main@@YAHXZ) [P:\Users\MEL\Desktop\Programming\CPP\c++11\build\Example.vcxproj]
[build] P:\Users\MEL\Desktop\Programming\CPP\c++11\build\Debug\Example.exe : fatal error LNK1120: 1 unresolved externals [P:\Users\MEL\Desktop\Programming\CPP\c++11\build\Example.vcxproj]
[proc] The command: "P:\Program Files\CMake\bin\cmake.EXE" --build p:/Users/MEL/Desktop/Programming/CPP/c++11/build --config Debug --target Example -j 18 -- exited with code: 1
[driver] Build completed: 00:00:01.659
[build] Build finished with exit code 1



![Screenshot 2024-08-11 213159](https://github.com/user-attachments/assets/ac1aca15-bf3d-40f6-a79f-788e625f6392)

**Comment by StarGate-One** [comment]
Did the `<vcpkg-root>\installed\<vcpkg-triplet>\lib\manual-link\SDL2main.lib` get linked - it is one of a few ports where manual linking of a library is required?

**Reply by Melio-1** [reply]
> Did the `<vcpkg-root>\installed\<vcpkg-triplet>\lib\manual-link\SDL2main.lib` get linked - it is one of a few ports where manual linking of a library is required?

No, I didn't linked it, i will try it now
